File: backend/routes/leaderboardRoutes.js
const express = require("express");
const router = express.Router();
const { getLeaderboard, clearLeaderboard } = require("../controllers/leaderboardController");

// Route to get the current leaderboard
router.get("/leaderboard", getLeaderboard);

// Optional: Route to clear the leaderboard (for admin purposes)
router.delete("/leaderboard", clearLeaderboard);

module.exports = router;

File: backend-golang/routes/leaderboard_routes.go
package routes

import (
	"backend-golang/controllers"

	"github.com/go-redis/redis/v8"
	"github.com/gorilla/mux"
)

func SetupLeaderboardRoutes(r *mux.Router, redisClient *redis.Client) {
	r.HandleFunc("/leaderboard", controllers.GetLeaderboard(redisClient)).Methods("GET")
	r.HandleFunc("/leaderboard", controllers.ClearLeaderboard(redisClient)).Methods("DELETE")
}
